### PR TITLE
test: Adding test for backup and restore keptn data

### LINKED ISSUE #6072: Create an integration test for backup and restore keptn data
## Action Item

Create an integration test that contains the logic of testing whether the backup and restore procedure described [here](https://keptn.sh/docs/0.10.x/operate/backup_and_restore/) is working.

## Acceptance Criteria
Integration tests contain tests that are testing backing up and restoring
ConfigurationService data as well as data stored in MongoDB

### PR DESCRIPTION
**Pull request by odubajDT**
Signed-off-by: odubajDT <ondrej.dubaj@dynatrace.com>

## This PR

- adds a new test for backuping and restoring keptn data

### Related Issues

Fixes #6072

### CODE REVIEW

**Comment by odubajDT** [comment]
Test written according to these steps: https://github.com/keptn/keptn.github.io/blob/e755bcaf07a0880a2c4bb307ab9ec0f709836745/content/docs/0.11.x/operate/backup_and_restore/index.md

File: test/go-tests/backuprestore_test.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+          tasks:
+            - name: "deployment"
+              properties:
+                deploymentstrategy: "direct"
+            - name: "release"
+`
+
+func Test_BackupRestore(t *testing.T) {
+	repoLocalDir := "../assets/podtato-head"
+	keptnProjectName := "podtato-head"
```
Comment (bacherfl) [inline test/go-tests/backuprestore_test.go:51]: please rename the project to some name that indicates what is tested , e.g. `backup-restore`

File: test/go-tests/init_test.go [hunk 2]
```diff
@@ -10,6 +10,7 @@ import (
 
 func TestMain(m *testing.M) {
 	if err := setup(); err != nil {
+		fmt.Printf("TestMain: error while setting up the tests")
```
Comment (bacherfl) [inline test/go-tests/init_test.go:13]: let's also add the error message here to provide more information

File: test/go-tests/backuprestore_test.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+	require.Nil(t, err)
+	err = os.Chdir("keptn-backup")
+	require.Nil(t, err)
+	err = os.MkdirAll("config-svc-backup", os.ModePerm)
+	require.Nil(t, err)
+
+	t.Logf("Executing backup of configuration-service")
+	configServicePod, err := ExecuteCommandf("kubectl get pods -n keptn -lapp.kubernetes.io/name=configuration-service -ojsonpath='{.items[0].metadata.name}'")
+	require.Nil(t, err)
+	configServicePod = removeQuotes(configServicePod)
```
Comment (bacherfl) [inline test/go-tests/backuprestore_test.go:105]: you can use `strings.Trim()` here instead of the `removeQuotes` function

**Comment by odubajDT** [comment]
Waiting on results of integration tests https://github.com/keptn/keptn/actions/runs/1503212194

**Comment by odubajDT** [comment]
Test seems to be working properly, @bacherfl can you please re-review the PR? https://github.com/keptn/keptn/actions/runs/1506686320 Thank you

**Review by bacherfl — APPROVED** [review]

**Pull request merged**